### PR TITLE
[AutoWarmth] optimisations and fixes to progressive warmth setting

### PR DESCRIPTION
**Pull request by zwim**
Fixes a quirk introduced with  #9252, which schedules not the actual warmth, but the next but one warmth when leaving suspend or standby.

Imperceptible to the user:
* Reduce calculations (save minimal power, esp. when wake from standby).
* Fixes stupid reschedules of `scheduleNextWarmthChanges` which may lead to 30.000(!) schedulings  if the reschedule time is below 1 second (as `os.date` only gives a second resolution). This might sometimes happen but very  seldom, but can be provoked wit a fixed schedule. This is not really noticeable to the user, no hang or something, but anyway it will get fixed with this PR....
_(For documentation: Set all times to 16:17:00; Solar noon to 16:17:10;  Sunset to 16:18:00; Civil dusk to 16:19:00; Nautical dusk to 16:20; Astronomical dusk to 18:18; Set system time to 16:17:50 and log executions of `scheduleNextWarmthChanges`)_

~~ToDo: Check corner cases around midnight ... simulations in the time machine seems good.  But I want to wait a few day with real life usage.~~

Real life test OK.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9442)

### CODE REVIEW

File: plugins/autowarmth.koplugin/main.lua [hunk 1]
```diff
@@ -324,6 +328,11 @@ function AutoWarmth:scheduleNextWarmthChange(time_s, search_pos, from_resume)
         if next_sched_time_s > 0 then
             -- This setWarmth will call scheduleNextWarmthChange which will schedule setWarmth again.
             UIManager:scheduleIn(next_sched_time_s, self.setWarmth, self, self.sched_warmths[self.sched_warmth_index], true)
+        else
+            -- ToDo: check if a schedule has to be in 1 sec or maybe later lets say at
```
Comment (Frenzie) [inline plugins/autowarmth.koplugin/main.lua:332]: You probably mean before merging, but in case not make it `--- @todo`.

Comment (zwim) [reply plugins/autowarmth.koplugin/main.lua:332]: Yes, as this is a draft PR, I meant I will rule this special case out before merging. (This might be a point in corner cases or not dunno now.)

Tust to be sure, I thought:
-- todo won't pass CI, so a developer won`t forget it, and others **_do_** see this.
--@todo is for later optimizations, so this is a "don't forget", "do it later" hint.

Am I right with these assumptions?

But anyway thanks for your care to point this out.

Comment (Frenzie) [reply plugins/autowarmth.koplugin/main.lua:332]: Yup, well, I don't think the first is typical but regardless. ^^

**Comment by zwim** [comment]
Are there reservations to a merge?

**Review by Frenzie — APPROVED** [review]
